x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java

```diff
@@ -330,7 +330,13 @@ public Sequence visitSequence(SequenceContext ctx) {
 
         // until is already parsed through sequenceTerm() above
         if (ctx.until != null) {
+            if (queries.size() == 2) {
+                throw new ParsingException(source, "A sequence requires a minimum of 2 queries (excluding UNTIL clause), found [1]");
+            }
             until = queries.remove(queries.size() - 1);
+            if (until.isMissingEventFilter()) {
+                throw new ParsingException(source, "UNTIL clause cannot be a negative clause (missing event)");
```

Comment:
The correct ParsingException would have been `ParsingException(until.source(), "UNTIL clause cannot be a....`